# Домашнее задание

В домашнем задании вам необходимо рассмтреть данные с олимпиады НТИ по профилю "Исскусственный интеллект" за 2019-2020 год.

Задачу можно найти по ссылке:

https://www.kaggle.com/c/onti-students-performance/data

## **Как загрузить данные в Google Colab?**

1. Из Github (файлы <25 МБ)
2. С локального диска
3. С Google Диска

In [1]:
# Github
# 1. Нажмите на файл с данными, которые хотите подключить
# 2. Нажмите "Raw"
# 3. Скопируйте ссылку открывшейся старницы
import pandas as pd

url1 = 'https://raw.githubusercontent.com/Adelaaas/Data_science_basic_22-23_1/main/class_work_4_Pandas/ONTI%20Students%20performance/X_train.csv'
url2 = 'https://raw.githubusercontent.com/Adelaaas/Data_science_basic_22-23_1/main/class_work_4_Pandas/ONTI%20Students%20performance/studs_info.csv'
url3 = 'https://raw.githubusercontent.com/Adelaaas/Data_science_basic_22-23_1/main/class_work_4_Pandas/ONTI%20Students%20performance/zachety.csv'
url4 = 'https://raw.githubusercontent.com/Adelaaas/Data_science_basic_22-23_1/main/class_work_4_Pandas/ONTI%20Students%20performance/y_train.csv'

X_train = pd.read_csv(url1)
studs_info = pd.read_csv(url2)
zachety = pd.read_csv(url3)
y_train = pd.read_csv(url4)

In [ ]:
# С локального компьютера

from google.colab import files
uploaded = files.upload()

In [ ]:
studs_info = pd.read_csv('studs_info.csv')
X_train = pd.read_csv('X_train.csv')
y_train = pd.read_csv('y_train.csv')
zachety = pd.read_csv('zachety.csv')

In [ ]:
# Google disk

from google.colab import drive
drive.mount('/content/drive')

In [ ]:
path = "copied path"
# /content/drive/MyDrive/Data Science для школьников/ОНТИ /ONTI Students performance/X_train.csv
df_bonus = pd.read_csv(path)

## Вариант 1.

Кто делает:
1. Александров Данила Александрович
2. Быков Дмитрий Александрович 
3. Горбунов Сергей Михайлович 
4. Горячко Дарья Романовна 
5. Громов Алексей Игоревич
6. Игнатов Константин Дмитриевич
7. Ямпольская Ирина Вадимовна
8. Дидковский Илья Радионович

1. Посмотреть размерность дата фреймов.
2. Соедить два датафрейма X_train и y_train в один train по индексам.
3. Посчитать пропуски в получившемся датафрейме train.
4. Посчитать сколько раз встречается каждая специальность в колонке studs_info['направление (специальность)'].
5. Создать новый датафрейм из studs_info, в котором будут только женщины с лечебного дела.
6. Разложить колонку "Дата выпуска" на три: Год выпуска, Месяц выпуска, День выпуска.
7. Посчитать сколько уникальных значений в колонке studs_info['Категория обучения'].
8. Заменить в колонке studs_info['Категория обучения'] все строковые категории на числовые.
9. Удалить ненужную на ваш вгляд колонку из датафрейма studs_info.
10. Посчитать сколько мужчин выпустились 2019-08-31.

In [75]:
# 1

print(studs_info.shape)
print(X_train.shape)
print(y_train.shape)
print(zachety.shape)

(5466, 16)
(105058, 8)
(105058, 2)
(133223, 8)


In [76]:
# 2 

train = pd.concat([X_train, y_train])
train

Unnamed: 0    STD_ID             НАПРАВЛЕНИЕ        ГОД  КУРС  \
0          51081.0  175711.0     38.03.02 Менеджмент  2018-2019   2.0   
1          72921.0  100647.0  31.05.01 Лечебное дело  2015-2016   3.0   
2          96438.0  199311.0   31.05.03 Стоматология  2019-2020   1.0   
3         273044.0  162002.0     38.03.02 Менеджмент  2017-2018   2.0   
4         371309.0  108384.0  31.05.01 Лечебное дело  2018-2019   5.0   
...            ...       ...                     ...        ...   ...   
105053         NaN       NaN                     NaN        NaN   NaN   
105054         NaN       NaN                     NaN        NaN   NaN   
105055         NaN       NaN                     NaN        NaN   NaN   
105056         NaN       NaN                     NaN        NaN   NaN   
105057         NaN       NaN                     NaN        NaN   NaN   

        СЕМЕСТР АТТЕСТАЦИЯ                                         ДИСЦИПЛИНА  \
0           4.0    Экзамен                                          Маркетинг   
1           5.0    Экзамен  Пропедевтика внутренних болезней, лучевая диаг...   
2           1.0    Экзамен                                              Химия   
3           3.0    Экзамен                                         Статистика   
4           9.0    Экзамен                                          Педиатрия   
...         ...        ...                                                ...   
105053      NaN        NaN                                                NaN   
105054      NaN        NaN                                                NaN   
105055      NaN        NaN                                                NaN   
105056      NaN        NaN                                                NaN   
105057      NaN        NaN                                                NaN   

           index  mark  
0            NaN   NaN  
1            NaN   NaN  
2            NaN   NaN  
3            NaN   NaN  
4            NaN   NaN  
...          ...   ...  
105053  224856.0   3.0  
105054  260998.0   3.0  
105055   57245.0   4.0  
105056  218529.0   4.0  
105057  233318.0   4.0  

[210116 rows x 10 columns]

In [77]:
studs_info

STD_ID     number Пол Статус Дата выпуска Категория обучения  \
0      99136    13951.0   Ж    ВЫП   2019-08-31                 БП   
1      99112    13998.0   Ж    ВЫП   2019-08-31                 БП   
2      99109    13999.0   Ж    ВЫП   2019-08-31                 БП   
3      99125    13953.0   Ж    ВЫП   2019-08-31                 БП   
4     100676    13950.0   М    ВЫП   2019-08-31                 БП   
...      ...        ...  ..    ...          ...                ...   
5461  100678      978.0   Ж     СТ          NaN                 БП   
5462  100663     1021.0   Ж     СТ          NaN                ДОГ   
5463  191630     6447.0   Ж     СТ          NaN                ДОГ   
5464  191625     3765.0   Ж     СТ          NaN                ДОГ   
5465  194978     5245.0   Ж     СТ          NaN                ДОГ   

     Форма обучения      Шифр направление (специальность)         Образование  \
0                 Д  31.05.01               Лечебное дело  2629.0          СР   
1                 Д  31.05.01               Лечебное дело  8240.0          СР   
2                 Д  31.05.01               Лечебное дело  2314.0          СР   
3                 Д  31.05.01               Лечебное дело  2345.0          СР   
4                 Д  31.05.01               Лечебное дело  2650.0          СР   
...             ...       ...                         ...     ...         ...   
5461              Д  31.05.01               Лечебное дело  2388.0          СР   
5462              Д  31.05.01               Лечебное дело  2554.0          СР   
5463              Д  31.05.01               Лечебное дело  6692.0          СР   
5464              Д  31.05.01               Лечебное дело     NaN         NaN   
5465              Д  31.05.01               Лечебное дело     NaN         NaN   

              Дата выдачи        Что именно закончил Год выпуска  \
0     2013-06-26 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
1     2012-06-23 00:00:00     МОУ СОШ N   2, ПЕСТОВО        2012   
2     2013-06-25 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
3     2013-06-23 00:00:00        Гимназия "ГАРМОНИЯ"        2013   
4     2012-06-29 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2012   
...                   ...                        ...         ...   
5461  2013-06-21 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
5462  2013-06-27 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
5463  2014-05-31 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2014   
5464                  NaN                        NaN         NaN   
5465                  NaN                        NaN         NaN   

     Месяц выпуска День выпуска  
0               06           26  
1               06           23  
2               06           25  
3               06           23  
4               06           29  
...            ...          ...  
5461            06           21  
5462            06           27  
5463            05           31  
5464           NaN          NaN  
5465           NaN          NaN  

[5466 rows x 16 columns]

In [78]:
# 3 

train.isna().sum()

Unnamed: 0     105058
STD_ID         105058
НАПРАВЛЕНИЕ    105058
ГОД            105058
КУРС           105058
СЕМЕСТР        105058
АТТЕСТАЦИЯ     105058
ДИСЦИПЛИНА     105058
index          105058
mark           105058
dtype: int64

In [79]:
# 4

studs_info['направление (специальность)'].value_counts()

Лечебное дело                                                                870
Юриспруденция                                                                462
Стоматология                                                                 452
Педагогическое образование (с двумя профилями подготовки)                    369
Педагогическое образование                                                   279
Экономика                                                                    226
Строительство                                                                214
Менеджмент                                                                   184
Лингвистика                                                                  136
Специальное (дефектологическое) образование                                  131
Теплоэнергетика и теплотехника                                               126
Журналистика                                                                 118
Конструкторско-технологическ

In [80]:
# 5

woman = studs_info[studs_info['Пол']=='Ж']
woman = woman[woman['направление (специальность)']=='Лечебное дело']
woman

STD_ID     number Пол Статус Дата выпуска Категория обучения  \
0      99136    13951.0   Ж    ВЫП   2019-08-31                 БП   
1      99112    13998.0   Ж    ВЫП   2019-08-31                 БП   
2      99109    13999.0   Ж    ВЫП   2019-08-31                 БП   
3      99125    13953.0   Ж    ВЫП   2019-08-31                 БП   
5      99099    13959.0   Ж    ВЫП   2019-08-31                 БП   
...      ...        ...  ..    ...          ...                ...   
5461  100678      978.0   Ж     СТ          NaN                 БП   
5462  100663     1021.0   Ж     СТ          NaN                ДОГ   
5463  191630     6447.0   Ж     СТ          NaN                ДОГ   
5464  191625     3765.0   Ж     СТ          NaN                ДОГ   
5465  194978     5245.0   Ж     СТ          NaN                ДОГ   

     Форма обучения      Шифр направление (специальность)         Образование  \
0                 Д  31.05.01               Лечебное дело  2629.0          СР   
1                 Д  31.05.01               Лечебное дело  8240.0          СР   
2                 Д  31.05.01               Лечебное дело  2314.0          СР   
3                 Д  31.05.01               Лечебное дело  2345.0          СР   
5                 Д  31.05.01               Лечебное дело  8800.0          СР   
...             ...       ...                         ...     ...         ...   
5461              Д  31.05.01               Лечебное дело  2388.0          СР   
5462              Д  31.05.01               Лечебное дело  2554.0          СР   
5463              Д  31.05.01               Лечебное дело  6692.0          СР   
5464              Д  31.05.01               Лечебное дело     NaN         NaN   
5465              Д  31.05.01               Лечебное дело     NaN         NaN   

              Дата выдачи        Что именно закончил Год выпуска  \
0     2013-06-26 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
1     2012-06-23 00:00:00     МОУ СОШ N   2, ПЕСТОВО        2012   
2     2013-06-25 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
3     2013-06-23 00:00:00        Гимназия "ГАРМОНИЯ"        2013   
5     2013-06-21 00:00:00     МОУ СОШ (полная), ХОЛМ        2013   
...                   ...                        ...         ...   
5461  2013-06-21 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
5462  2013-06-27 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
5463  2014-05-31 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2014   
5464                  NaN                        NaN         NaN   
5465                  NaN                        NaN         NaN   

     Месяц выпуска День выпуска  
0               06           26  
1               06           23  
2               06           25  
3               06           23  
5               06           21  
...            ...          ...  
5461            06           21  
5462            06           27  
5463            05           31  
5464           NaN          NaN  
5465           NaN          NaN  

[564 rows x 16 columns]

In [81]:
# 6

def my_split_year(line):
  if type(line) == str:
    return line.split('-')[0]
  return line

def my_split_month(line):
  if type(line) == str:
    try:
      return line.split('-')[1]
    except:
      return line
  return line

def my_split_day(line):
  if type(line) == str:
    try:
      mass = line.split(' ')[0]
      return mass.split('-')[2]
    except:
      return line
  return line

# 2013-06-26 00:00:00
studs_info['Год выпуска'] = studs_info['Дата выдачи'].apply(my_split_year)
studs_info['Месяц выпуска'] = studs_info['Дата выдачи'].apply(my_split_month)
studs_info['День выпуска'] = studs_info['Дата выдачи'].apply(my_split_day)
studs_info

STD_ID     number Пол Статус Дата выпуска Категория обучения  \
0      99136    13951.0   Ж    ВЫП   2019-08-31                 БП   
1      99112    13998.0   Ж    ВЫП   2019-08-31                 БП   
2      99109    13999.0   Ж    ВЫП   2019-08-31                 БП   
3      99125    13953.0   Ж    ВЫП   2019-08-31                 БП   
4     100676    13950.0   М    ВЫП   2019-08-31                 БП   
...      ...        ...  ..    ...          ...                ...   
5461  100678      978.0   Ж     СТ          NaN                 БП   
5462  100663     1021.0   Ж     СТ          NaN                ДОГ   
5463  191630     6447.0   Ж     СТ          NaN                ДОГ   
5464  191625     3765.0   Ж     СТ          NaN                ДОГ   
5465  194978     5245.0   Ж     СТ          NaN                ДОГ   

     Форма обучения      Шифр направление (специальность)         Образование  \
0                 Д  31.05.01               Лечебное дело  2629.0          СР   
1                 Д  31.05.01               Лечебное дело  8240.0          СР   
2                 Д  31.05.01               Лечебное дело  2314.0          СР   
3                 Д  31.05.01               Лечебное дело  2345.0          СР   
4                 Д  31.05.01               Лечебное дело  2650.0          СР   
...             ...       ...                         ...     ...         ...   
5461              Д  31.05.01               Лечебное дело  2388.0          СР   
5462              Д  31.05.01               Лечебное дело  2554.0          СР   
5463              Д  31.05.01               Лечебное дело  6692.0          СР   
5464              Д  31.05.01               Лечебное дело     NaN         NaN   
5465              Д  31.05.01               Лечебное дело     NaN         NaN   

              Дата выдачи        Что именно закончил Год выпуска  \
0     2013-06-26 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
1     2012-06-23 00:00:00     МОУ СОШ N   2, ПЕСТОВО        2012   
2     2013-06-25 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
3     2013-06-23 00:00:00        Гимназия "ГАРМОНИЯ"        2013   
4     2012-06-29 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2012   
...                   ...                        ...         ...   
5461  2013-06-21 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
5462  2013-06-27 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
5463  2014-05-31 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2014   
5464                  NaN                        NaN         NaN   
5465                  NaN                        NaN         NaN   

     Месяц выпуска День выпуска  
0               06           26  
1               06           23  
2               06           25  
3               06           23  
4               06           29  
...            ...          ...  
5461            06           21  
5462            06           27  
5463            05           31  
5464           NaN          NaN  
5465           NaN          NaN  

[5466 rows x 16 columns]

In [82]:
# 7

len(studs_info['Категория обучения'].unique())

3

In [83]:
# 8

keys = {}
for i, item in enumerate(studs_info['Категория обучения'].unique()):
  keys[item] = i + 1

studs_info['Категория обучения'].apply(lambda x: keys[x])

studs_info

STD_ID     number Пол Статус Дата выпуска Категория обучения  \
0      99136    13951.0   Ж    ВЫП   2019-08-31                 БП   
1      99112    13998.0   Ж    ВЫП   2019-08-31                 БП   
2      99109    13999.0   Ж    ВЫП   2019-08-31                 БП   
3      99125    13953.0   Ж    ВЫП   2019-08-31                 БП   
4     100676    13950.0   М    ВЫП   2019-08-31                 БП   
...      ...        ...  ..    ...          ...                ...   
5461  100678      978.0   Ж     СТ          NaN                 БП   
5462  100663     1021.0   Ж     СТ          NaN                ДОГ   
5463  191630     6447.0   Ж     СТ          NaN                ДОГ   
5464  191625     3765.0   Ж     СТ          NaN                ДОГ   
5465  194978     5245.0   Ж     СТ          NaN                ДОГ   

     Форма обучения      Шифр направление (специальность)         Образование  \
0                 Д  31.05.01               Лечебное дело  2629.0          СР   
1                 Д  31.05.01               Лечебное дело  8240.0          СР   
2                 Д  31.05.01               Лечебное дело  2314.0          СР   
3                 Д  31.05.01               Лечебное дело  2345.0          СР   
4                 Д  31.05.01               Лечебное дело  2650.0          СР   
...             ...       ...                         ...     ...         ...   
5461              Д  31.05.01               Лечебное дело  2388.0          СР   
5462              Д  31.05.01               Лечебное дело  2554.0          СР   
5463              Д  31.05.01               Лечебное дело  6692.0          СР   
5464              Д  31.05.01               Лечебное дело     NaN         NaN   
5465              Д  31.05.01               Лечебное дело     NaN         NaN   

              Дата выдачи        Что именно закончил Год выпуска  \
0     2013-06-26 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
1     2012-06-23 00:00:00     МОУ СОШ N   2, ПЕСТОВО        2012   
2     2013-06-25 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
3     2013-06-23 00:00:00        Гимназия "ГАРМОНИЯ"        2013   
4     2012-06-29 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2012   
...                   ...                        ...         ...   
5461  2013-06-21 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
5462  2013-06-27 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
5463  2014-05-31 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2014   
5464                  NaN                        NaN         NaN   
5465                  NaN                        NaN         NaN   

     Месяц выпуска День выпуска  
0               06           26  
1               06           23  
2               06           25  
3               06           23  
4               06           29  
...            ...          ...  
5461            06           21  
5462            06           27  
5463            05           31  
5464           NaN          NaN  
5465           NaN          NaN  

[5466 rows x 16 columns]

In [86]:
# 9

studs_info.drop(['Форма обучения'], axis=1)
studs_info

STD_ID     number Пол Статус Дата выпуска Категория обучения  \
0      99136    13951.0   Ж    ВЫП   2019-08-31                 БП   
1      99112    13998.0   Ж    ВЫП   2019-08-31                 БП   
2      99109    13999.0   Ж    ВЫП   2019-08-31                 БП   
3      99125    13953.0   Ж    ВЫП   2019-08-31                 БП   
4     100676    13950.0   М    ВЫП   2019-08-31                 БП   
...      ...        ...  ..    ...          ...                ...   
5461  100678      978.0   Ж     СТ          NaN                 БП   
5462  100663     1021.0   Ж     СТ          NaN                ДОГ   
5463  191630     6447.0   Ж     СТ          NaN                ДОГ   
5464  191625     3765.0   Ж     СТ          NaN                ДОГ   
5465  194978     5245.0   Ж     СТ          NaN                ДОГ   

     Форма обучения      Шифр направление (специальность)         Образование  \
0                 Д  31.05.01               Лечебное дело  2629.0          СР   
1                 Д  31.05.01               Лечебное дело  8240.0          СР   
2                 Д  31.05.01               Лечебное дело  2314.0          СР   
3                 Д  31.05.01               Лечебное дело  2345.0          СР   
4                 Д  31.05.01               Лечебное дело  2650.0          СР   
...             ...       ...                         ...     ...         ...   
5461              Д  31.05.01               Лечебное дело  2388.0          СР   
5462              Д  31.05.01               Лечебное дело  2554.0          СР   
5463              Д  31.05.01               Лечебное дело  6692.0          СР   
5464              Д  31.05.01               Лечебное дело     NaN         NaN   
5465              Д  31.05.01               Лечебное дело     NaN         NaN   

              Дата выдачи        Что именно закончил Год выпуска  \
0     2013-06-26 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
1     2012-06-23 00:00:00     МОУ СОШ N   2, ПЕСТОВО        2012   
2     2013-06-25 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
3     2013-06-23 00:00:00        Гимназия "ГАРМОНИЯ"        2013   
4     2012-06-29 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2012   
...                   ...                        ...         ...   
5461  2013-06-21 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
5462  2013-06-27 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2013   
5463  2014-05-31 00:00:00  ИНОГОРОДНЯЯ СРЕДНЯЯ ШКОЛА        2014   
5464                  NaN                        NaN         NaN   
5465                  NaN                        NaN         NaN   

     Месяц выпуска День выпуска  
0               06           26  
1               06           23  
2               06           25  
3               06           23  
4               06           29  
...            ...          ...  
5461            06           21  
5462            06           27  
5463            05           31  
5464           NaN          NaN  
5465           NaN          NaN  

[5466 rows x 16 columns]

In [90]:
# 10

man = studs_info[studs_info['Пол']=='М']
man = man[man['Год выпуска']=='2019']
man = man[man['Месяц выпуска']=='08']
man = man[man['День выпуска']=='31']
len(man)

0

## Вариант 2.

Кто делает:
1. Ивлиев Степан Дмитриевич
2. Игнатов Игорь Дмитриевич
3. Казанов Амир Ринатович
4. Клячин Иван Николаевич
5. Котов Александр Андреевич
6. Мелешкин Кирилл Андреевич
7. Файзуллин Илья Денисович
8. Чупрова Ася Александровна

1. Посмотреть размерность дата фреймов.
2. Соедить два датафрейма X_train и y_train в один train по индексам.
3. Посчитать пропуски в датафрейме studs_info.
4. Посчитать сколько мужчин, сколько женщин в датафрейме studs_info.
5. Создать новый датафрейм из studs_info, в котором будут только мужчины 2019-06-30 года выпуска.
6. Разложить колонку "Дата выдачи" на четыре: Год выдачи, Месяц выдачи, День выдачи, Время выдачи.
7. Посчитать сколько уникальных значений в колонке studs_info['Статус'] и сколько раз каждое значение встречается в датафрейм.
8. Посчитать сколько раз какая оценка встречатся в датафрейм y_train.
9. Посчитать средний балл для студента с ID=175711 по всем предметам.
10. Удалить любую ненужны колонку на ваш вгляд из датафрейм train.

In [ ]:
# Ваше решение

df[df['name'] == 'Ivan']

## Вариант 3.

Кто делает:
1. Морозов Максим Олегович
2. Переттруов Никита Артемович
3. Пославский Егор Константинович
4. Садохин Константин Павлович
5. Скиданов Дмитрий Артёмович
6. Судариков Андрей Геннадьевич
7. Таланов Максим Андреевич

Задания:
1. Посмотреть размерность дата фреймов.
2. Соедить два датафрейма X_train и y_train в один train по индексам.
3. Посчитать пропуски в датафрейме studs_info.
4. Посчитать сколько уникальных значений в колонке АТТЕСТАЦИЯ и сколько раз каждое из них встречается.
5. Переименовать все колонки датафрейм train, чтобы названия были маленькими буквами.
6. Разложить колонку "направление" на две: Код направления, Название направления.
7. Создать новый датафрейм, в котором будут только мужчины 3 курса обучения.
8. Заменить все строковые значения в колонке Пол на числовые.
9. Сколько всего студентов сдавали предмет "Информатика" в виде экзамена? И какой средний балл по всем сдававшим?

In [ ]:
# Ваше решение